enhancements/microshift/kubernetes-for-industrial-edge.md

```diff
@@ -0,0 +1,435 @@
+node-side configuration will be done via the standard kubelet
+configuration file.
+
+Other operating system services will be configured outside of
+MicroShift using their standard configuration files. There is no
+machine-config-operator.
+
+Because MicroShift will be treated as a single application, we will
+not expose separate health checks or implement per-component restart
+features. If any embedded component is unhealthy, the application will
+be treated as unhealthy and be restarted.
+
+### Workflow Description
+
+#### Deploying
+
+Application deployers will use RHEL for Edge tools such as Image
+Builder to construct full system images containing RHEL, MicroShift,
+agents, custom operating system configuration, data prerequisites, and
+optionally application images. Those system images will be installed
+on the target devices either individually for development/testing or
+via manufacturer imaging processes for large-scale production
+deployments. RHEL for Edge uses ostree-based images, and an [example
+workflow](https://github.com/openshift/microshift/blob/main/docs/rhel4edge_iso.md)
+for constructing a basic image is described in the MicroShift
+documentation. We are not creating a separate installer for
+MicroShift.
+
+#### Upgrading
+
+Because the operating system is based on rpm-ostree, upgrades are
+applied either by re-imaging the device, or by using the ostree update
+process and rebooting into a new ostree. The ostree images can be
+updated by downloading only the differences, which satisfies the
+requirement to minimize over-the-wire bandwidth costs of upgrades.
+
+An [example workflow illustrating
+upgrades](https://github.com/redhat-et/microshift-demos/tree/main/ostree-demo)
+is described in the MicroShift demos repository.
+
+There is no cluster-version-operator. MicroShift is not
+self-upgrading, although another agent on the host may participate in
+orchestrating upgrades.
+
+#### Configuring
+
+MicroShift reads its configuration file once on startup. Changes to
+the file will only be implemented after the application is
+restarted. Most configuration changes in production settings would be
+rolled out as new ostree images.
+
+### API Extensions
+
+MicroShift does not add any APIs to OpenShift.
+
+### Implementation Details/Notes/Constraints [optional]
+
+### Risks and Mitigations
+
+If an upgrade fails, the RHEL for Edge greenboot feature will allow
+the system to reboot back into the older ostree.
+
+Building a single binary means that secondary dependencies of
+components need to be kept compatible. This is especially problematic
+during a kubernetes rebase in Openhift, but applies to many other
+dependencies as well. We are working with the control plane team to
+find ways to mitigate this risk, including moving some code out of the
+OpenShift API server into controllers that would not depend on
+kubernetes at all.
+
+Building a different form factor downstream of OpenShift means it is
```

Comment:
its more accurate to say its downstream of the release image since both hosted/standalone models share that release image.

Reply:
clarified